mazeRunner/src/mazeRunner.js

```diff
@@ -0,0 +1,68 @@
+const mazeRunner = (maze, directions) => {
+  const isStartPoint = (element) => element == 2;
+  // returns an array of original starting position
+  const startingPosition = (maze) => {
+    for (let i = 0; i < maze.length; i++) {
+      //console.log(Array.isArray(maze[i]));
+      startPointJ = maze[i].findIndex(isStartPoint);
+      if (startPointJ > -1) {
+        return [i, startPointJ];
+      }
+    }
+  };
+
+  // takes a direction (string) and position (array) and returns an array
+  const newPosition = (direction, position) => {
+    switch (direction) {
+      case "E":
+        return [position[0], position[1] + 1];
+      case "W":
+        return [position[0], position[1] - 1];
+      case "N":
+        return [position[0] - 1, position[1]];
+      case "S":
+        return [position[0] + 1, position[1]];
+    }
+  };
+
+  //console.log("original position of the 2", startingPosition(maze));
+
+  const result = (maze, position) => {
+    // console.log("position in result", position);
+    // console.log("maze point", maze[position[0]][position[1]]);
+    if (
+      maze[position[0]][position[1]] === 0 ||
+      maze[position[0]][position[1]] === 2
+    ) {
+      return "Lost";
+    } else if (maze[position[0]][position[1]] === 1) {
+      return "Dead";
+    } else if (maze[position[0]][position[1]] === 3) {
+      return "Finish";
+    }
+  };
+
+  for (let i = 0; i < directions.length; i++) {
```

Comment:
You could use a `for...of` loop here if you wanted instead. Using the counter method has become a bit less popular.
```suggestion
  for (const direction of directions)  {
  ...
```
Then you can just use `direction` any place you used `directions[i]`. Just means you don't have to think about that `i` variable.

The only other change you'd have to make would be around the
```
if (i === 0) {
      position = startingPosition(maze);
}
```
Can you think of how you could change the code to still have that?
(Hint: logically where else could it run?)

Reply:
thanks, i will look at for of!  will refactor and send to you again :D thanks so much